src/get-started/first-java-service.md

```diff
@@ -0,0 +1,615 @@
+
+### Service Instantiation
+
+#### Install the Java Launcher
+
+To register a service artifact that we built previously in the network,
+we will need `exonum-launcher` tool. It is a Python application which we
+recommend to install in a [virtual environment][python-venv]:
+
+```shell
+python3 -m venv .venv
+source .venv/bin/activate
+```
+
+[python-venv]: https://docs.python.org/3/library/venv.html
+
+Then install the `exonum-launcher` with the Java runtime support:
+
+```shell
+pip install exonum-launcher-java-plugins
+```
+
+Check it works:
+
+```shell
+python -m exonum_launcher --help
+```
+
+#### Start a Test Instance
+
+Next, we shall place the service artifact into an artifacts directory
+of the node: `testnet/artifacts`.
+
+```shell
+# Create the artifacts directory
+mkdir testnet/artifacts
+# Copy the artifact
+cp car-registry-service/target/car-registry-service-1.0.0-SNAPSHOT-artifact.jar \
+   testnet/artifacts/
+```
+
+<!--
+TODO: Shall the _node_ (= Java runtime) create an artifacts directory if one
+does not exist already? If it shall, won't it cause problems if we launch
+several nodes locally with the same (not yet existing) artifacts directory?
+
+It is somewhat annoying to always have to create the dir :-)
+-->
+
+<!--
+TODO: Shall we place the burden of copying (= uploading the JAR)
+on the launcher-plugin + Java runtime pair?
+-->
+
+Launch the service:
+
+```shell
+python -m exonum_launcher -i deploy-start-config.yml
+```
+
+Launcher will take the service instance name and other parameters from
+the configuration file, and submit the request to the node.
+The launcher must print the status of the service artifact deploy
+and the service instance start.
+We can also verify that both operations succeeded via the node API:
+
+```shell
+curl -s http://127.0.0.1:3000/api/system/v1/services | jq
+```
+
+#### Invoke the Service Operations
+
+We will use a light client application to invoke the service operations.
+Development of service client applications is not covered in this tutorial,
+but the client for the car registry is provided in the tutorial
+repository as a third module, `car-registry-client`.
+
+If you have not cloned the repository already, clone it and build the client:
+
+```shell
+git clone git@github.com:exonum/exonum-java-binding.git
+cd exonum-java-binding/exonum-java-binding/tutorials/car-registry
+mvn package -pl car-registry-client -am
+```
+
+Check it is built successfully:
+
+```shell
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar -h
+```
+
+First, generate an Ed25519 key pair, that the client will use to sign
+the transactions to our service:
+
+```shell
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  keygen
+```
+
+Now, try to add your own vehicles to the blockchain:
+
+```shell
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  add-vehicle -a -n=test-car-registry "My car" "VW" "Polo" "$USER"
```

Comment:
I think it could be worth explaining what `-a` and `-n` arguments mean.

Reply:
I added, but, just in case, these command also have help:

```
(.venv) dmitry-mac:car-registry user$ java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar   change-owner -a -n=test-car-registry "Car 2" "$USER" -h
Usage: <main class> change-owner (-n=<name> | -i=<id>) [-ahV] <vehicleId>
                                 <newOwner>
Changes the owner of the vehicle with the given ID.
      <vehicleId>
      <newOwner>
  -a, --await             Await the transaction commit.
                          The transaction execution result will be printed on
                            the console.
  -h, --help              Show this help message and exit.
  -i, --service-id=<id>   The numeric service instance ID, assigned on the
                            instance start.
  -n, --service-name=<name>
                          The service instance name.
  -V, --version           Print version information and exit.
```